pkgs/development/compilers/avian/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchurl, zlib, openjdk }:
+
+stdenv.mkDerivation {
+  name = "avian-1.2.0";
+
+  src = fetchurl {
+    url = https://readytalk.github.io/avian-web/avian-1.2.0.tar.bz2;
+    sha256 = "0y4qldsz2s99vyax67bif6d3li2wnbvmpiba598bx4rn2idnzk7f";
+  };
+
+  buildInputs = [
+    zlib
+    openjdk
+  ];
+
+  buildPhase = ''
+    make
+  '';
+
+  installPhase = ''
+    mkdir -p $out/bin
+    cp build/*/avian $out/bin/
+    cp build/*/avian-dynamic $out/bin/
+  '';
+
+  meta = {
+    description = "Lightweight Java virtual machine";
+    longDescription = ''
+      Avian is a lightweight virtual machine and class library designed
+      to provide a useful subset of Java’s features, suitable for
+      building self-contained applications.
+    '';
+    homepage = https://readytalk.github.io/avian/;
+    license = stdenv.lib.licenses.isc;
+    platforms = stdenv.lib.platforms.all;
```

Comment:
I think `all` is appropriate here, based on the architectures listed in the [avian readme](https://github.com/ReadyTalk/avian/blob/d9650be5701becf7b5d2d38b48f445c655d8d04f/README.md).